### DISCUSSION TITLE
Drift files vs Dart code

### DISCUSSION THREAD

**Discussion by ycherniavskyi** [simolus3/drift#1513] [General]
What are the preferable way to create a database structure and queries in Drift?

Will Drift files and Dart code be always interchangeable and can be used both separately and together?

Just curious what technic is better to use in a new project?

**Comment by simolus3** [comment]
Some very minor features may only be available in Drift files or only available in Dart code. In general, they can be used interchangeably though (Dart code can write queries for Drift-defined tables, and vice-versa).

My recommendation would be to start using the format you're more comfortable with:

- Do you need to write very advanced queries or care about optimized SQL? Drift files are probably the way to do.
- Do you have lots of simple CRUD-type queries? Writing them in Dart might be simpler.

But I think that personal preferences should weight more than that. If you like writing SQL, Drift files are the tool for you! If prefer writing Dart, you can do that too! Both options are first-class APIs and are designed to complement each other.

**Reply by ycherniavskyi** [reply]
Thank you for the clarification. I prefer Dart but was afraid that in the future this technic will be abandoned.

**Comment by kuhnroyal** [comment]
From my experience you later have to write some SQL for table migrations anyways so I found it easier to directly start with Drift-files.
